test/test_sbol2_sbol3_direct.py

```diff
@@ -149,5 +149,101 @@ def test_2to3_collection_conversion(self):
             self.assertFalse(file_diff(str(tmp2), str(TEST_FILES / 'sbol_3to2_collection.xml')))
     '''
 
+    def test_empty_componentDefinition(self):
+            """Test ability to convert a simple part from SBOL2 to SBOL3"""
+            # Load an SBOL2 document and check its contents
+            doc2 = sbol2.Document()
+            doc2.read(TEST_FILES / 'empty_componentDefinition.xml')
+            # Convert to SBOL3 and check contents
+            doc3 = convert2to3(doc2, use_native_converter=True)
+            self.assertEqual(len(doc3.validate()), 0)
+            with tempfile.TemporaryDirectory() as tmpdir:
+                tmp3 = Path(tmpdir) / 'doc3.nt'
+                doc3.write(tmp3)
+                self.assertFalse(file_diff(str(tmp3), str(TEST_FILES / 'empty_componentDefinition.nt')))
+                doc2_loop = convert3to2(doc3, True)
+                # report = doc2.validate()
+                # self.assertEqual(len(report), 0, f'Validation failed: {report}')
+                tmp2 = Path(tmpdir) / 'doc2_loop.xml'
+                doc2_loop.write(tmp2)
+                self.assertFalse(file_diff(str(tmp2), str(TEST_FILES / 'empty_componentDefinition.xml')))
+
+            # Do the same for a file without versioning    
+            doc2 = sbol2.Document()
+            doc2.read(TEST_FILES / 'empty_componentDefinition_nover.xml')
+            # Convert to SBOL3 and check contents
+            doc3 = convert2to3(doc2, use_native_converter=True)
+            self.assertEqual(len(doc3.validate()), 0)
+            with tempfile.TemporaryDirectory() as tmpdir:
+                tmp3 = Path(tmpdir) / 'doc3.nt'
+                doc3.write(tmp3)
+                self.assertFalse(file_diff(str(tmp3), str(TEST_FILES / 'empty_componentDefinition_nover.nt')))
+                doc2_loop = convert3to2(doc3, True)
+                # report = doc2.validate()
+                # self.assertEqual(len(report), 0, f'Validation failed: {report}')
+                tmp2 = Path(tmpdir) / 'doc2_loop.xml'
+                doc2_loop.write(tmp2)
+                self.assertFalse(file_diff(str(tmp2), str(TEST_FILES / 'empty_componentDefinition_nover.xml')))
+                
+    def test_minimal_sbol2_collection(self):
+            """Test ability to convert a collection of simple parts from SBOL2 to SBOL3"""
+            # Depends on working Component Definition conversion
+            # Load an SBOL2 document and check its contents
+            doc2 = sbol2.Document()
+            doc2.read(TEST_FILES / 'minimal_sbol2_collection.xml')
+            # Convert to SBOL3 and check contents
+            doc3 = convert2to3(doc2, use_native_converter=True)
+            self.assertEqual(len(doc3.validate()), 0)
+            with tempfile.TemporaryDirectory() as tmpdir:
+                tmp3 = Path(tmpdir) / 'doc3.nt'
+                doc3.write(tmp3)
+                #self.assertFalse(file_diff(str(tmp3), str(TEST_FILES / 'BBa_J23101_patched.nt')))
```

Comment:
Please add the equivalent SBOL3 `.nt` file to the tests and check it here. You can get it from an instance of `doc3.nt` produced by a test run.